src/python/pants/backend/jvm/subsystems/jvm_tool_mixin.py

```diff
@@ -183,42 +184,36 @@ def register_jvm_tool_decl(cls, register, decl, **kwargs):
 
   class CombinedJvmToolsError(JvmToolDeclError): pass
 
-  _combined_tools = []
-
-  @classmethod
-  def register_combined_jvm_tools(cls, register, tool_decls):
-    """Register `JvmToolDeclaration`s as JVM tools, which can be invoked as a single jar.
+  @classproperty
+  def combined_jvm_tool_names(cls):
+    """Return a list of tool names which can be invoked as a single jar.
 
     This allows tools to be invoked one at a time, but with the combined classpath of all of them
     using ensure_combined_jvm_tool_classpath(). This allows creating nailgun instances for the
     tools which have the same fingerprint, and allows a task to invoke multiple different JVM tools
-    from the same nailgun instances.
+    from the same nailgun instances. See #7089.
     """
-    for decl in tool_decls:
-      cls.register_jvm_tool_decl(register, decl)
-      cls._combined_tools.append(decl)
+    raise NotImplementedError(
+      'combined_jvm_tool_names must be implemented to use multiple JVM tools '
```

Comment:
Can you phrase this more like "is not implemented, see issue #7089", rather than "must be implemented" - the latter suggests to me that some subclass should be implementing, whereas the former suggests pants has a TODO